tests/unit/api/bucketGetLocation.js

```diff
@@ -0,0 +1,87 @@
+import assert from 'assert';
+
+import bucketPut from '../../../lib/api/bucketPut';
+import bucketGetLocation from '../../../lib/api/bucketGetLocation';
+import { cleanup,
+    DummyRequestLogger,
+    makeAuthInfo }
+from '../helpers';
+import config from '../../../lib/Config';
+
+const log = new DummyRequestLogger();
+const authInfo = makeAuthInfo('accessKey1');
+const bucketName = 'bucketGetLocationTestBucket';
+
+const testBucketPutRequest = {
+    bucketName,
+    headers: { host: `${bucketName}.s3.amazonaws.com` },
+    url: '/',
+};
+
+const testGetLocationRequest = {
+    bucketName,
+    headers: {
+        host: `${bucketName}.s3.amazonaws.com`,
+    },
+    url: '/?location',
+    query: { location: '' },
+};
+
+// test for old and new config
+const locationConstraints = config.locationConstraints ||
+{ foo: 'foo', toto: 'toto' };
+
+describe('getBucketLocation API', () => {
+    Object.keys(locationConstraints).forEach(location => {
+        if (location === 'us-east-1') {
+            // if region us-east-1 should return empty string
+            // see next test.
+            return;
+        }
+        describe(`with ${location} LocationConstraint`, () => {
+            beforeEach(done => {
+                cleanup();
+                bucketPut(authInfo, testBucketPutRequest,
+                location, log, done);
+            });
+            afterEach(() => cleanup());
+            it(`should return ${location} LocationConstraint xml`, done => {
+                bucketGetLocation(authInfo, testGetLocationRequest, log,
+                (err, res) => {
+                    if (err) {
+                        process.stdout.write(`Err putting cors config ${err}`);
```

Comment:
`Err getting location constraint`, or you can `assert.strictEqual(err, null, 'Unexpected ${err} getting location constraint')` except using backticks for the quotemarks